### PR TITLE
Fix typescript error in i18n

### PR DESCRIPTION
**Pull request by alex-page**
Updated `NumberFormatOptionsStyleRegistry` to align with latest version of typescript.

### CODE REVIEW

File: packages/react-i18n/src/i18n.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ import {
 } from './utilities';
 
 export interface NumberFormatOptions extends Intl.NumberFormatOptions {
-  as?: 'number' | 'currency' | 'percent';
+  as?: 'currency' | 'decimal' | 'percent' | 'unit';
```
[the remark this answers is not in the document]
Comment (BPScott) [reply packages/react-i18n/src/i18n.ts:57]: Probably nitpicky but I'd be tempted to go for `as?: 'decimal' | 'currency' | 'percent';`

That way we're a subset of the possible options.

**Review by BPScott — COMMENTED** [review]
Can we update this to `as?: 'decimal' | 'currency' | 'percent';` so we fix the single problematic value, rather than widening the set of allowed values to include the es2020+ only 'unit'.

---

In TS 5.4 `Intl.NumberFormatOptions['style']` was `string | undefined`.

TS 5.5 strictens that up to be a set of possible values (`keyof Intl.NumberFormatOptionsStyleRegistry`, the exact values of which change depending on which es version you're targeting - es5 supports decimal, percent and currency. es2020 adds unit).

No idea why `/snapit` is installing TS 5.5 when we request ts 5.4 in the package.json though.

---

I'm not sure if we want to count this as a patch or major.

We're removing a "valid" value which is technically a breaking change

But [MDN suggests that `number` was never a valid value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#style_options) for the style option in the first place so I suspect that trying to use 'number" would have resulted in a runtime error so I suspect nobody was ever actually using that for realisies anyway.

I'm tempted to try this as a patch change, do a snapit and see what breaks in web - if that's all good then do a patch.

**Comment by alex-page** [comment]
Tophatted in `/web` in https://github.com/Shopify/web/pull/138778

**Review by BPScott — APPROVED** [review]

**Pull request merged**